src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1164,6 +1164,31 @@ public void updateFeedbackSession(FeedbackSessionAttributes newSession)
 
         fsDb.updateFeedbackSession(newSession);
     }
+
+    /**
+     * Decrease the response rate of feedback session by 1 if the deleted response is the last response from
+     * the particular student in that feedback session; for use after deleting a student response
+     * @param studentEmail
+     * @param sessionName
+     * @param courseId
+     */
+    public void updateSessionResponseRateForDeletedStudentResponse(String studentEmail, String sessionName,
+            String courseId) throws InvalidParametersException, EntityDoesNotExistException {
+        if (!hasRemainingResponsesFromStudent(studentEmail, sessionName, courseId)) {
+            deleteStudentFromRespondentList(studentEmail, sessionName, courseId);
+        }
+    }
```

Comment:
@damithc, yes, this is susceptible to eventual consistency, but it is hard to see how we will be able to update the response rate otherwise whenever we delete a response through an instructor's action. 

@YongJieYongJie, Maybe one possible solution is if `hasRemainingResponsesFromStudent` is made public instead. The cascade is still handled by `deleteStudentFromRespondentList`. We can remove this `updateSessionResponseRateForDeletedStudentResponse` method and just inline it in `FeedbackResponsesLogic`